storage/opaque-origin.https.html

```diff
@@ -0,0 +1,78 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>StorageManager API and opaque origins</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+
+function load_iframe(src, sandbox) {
+  return new Promise(resolve => {
+    const iframe = document.createElement('iframe');
+    iframe.onload = () => { resolve(iframe); };
+    if (sandbox)
+      iframe.sandbox = sandbox;
+    iframe.srcdoc = src;
+    iframe.style.display = 'none';
+    document.documentElement.appendChild(iframe);
+  });
+}
+
+function wait_for_message(iframe) {
+  return new Promise(resolve => {
+    self.addEventListener('message', function listener(e) {
+      if (e.source === iframe.contentWindow) {
+        resolve(e.data);
+        self.removeEventListener('message', listener);
+      }
+    });
+  });
+}
+
+function make_script(snippet) {
+  return '<script>' +
+         '  window.onmessage = () => {' +
+         '    try {' +
+         '      (' + snippet + ')' +
+         '        .then(' +
+         '          result => {' +
+         '            window.parent.postMessage({result: "no rejection"}, "*");' +
+         '          }, ' +
+         '          error => {' +
+         '            window.parent.postMessage({result: error.name}, "*");' +
+         '          });' +
+         '    } catch (ex) {' +
+         '      window.parent.postMessage({result: ex.message}, "*");' +
```

Comment:
If the API isn't present e.g. run in a non-SecureContext, or just not implemented.

Included since it's better than a timeout.